hail/src/main/scala/is/hail/expr/ir/functions/Functions.scala

```diff
@@ -296,6 +296,27 @@ abstract class RegistryFunctions {
     })
   }
 
+  def registerCodeWithType(mname: String, aTypes: Array[Type], rType: Type, pt: (Type, Seq[PType]) => PType)
```

Comment:
I don't love this name. There's also a way to do this without registering another function -- on the `registerCode` that takes a Seq, add a `computeReturnPhysicalType` argument with signature `(Type, Seq[PType]) => PType`. The precedence is use this function if not null, then use `pt` if not null, then use canonical virtual type (which will go away soon)

Reply:
Got it, will take care of in the morning, thank you

Reply:
also, feel free to push back if you start prototyping this interface and it's terrible.

Reply:
Would you be open to changing this function signature throughout? The "calculate ptype" anonymous function differs in signature from `returnPType` (in missing the returnType argument), so it's not possible to perform the same work. I think those signatures should be the same.

Proposal: change all registerCode and registerCodeWithMissingness function signatures from `pt: (PType*) => PType` to `pt: (Type, PType*) => PType`. 

If you like that idea, we keep this version, and I follow up with a PR that changes these signatures, after I solve the random seeded function issue.

Reply:
yeah, changing it everywhere seems fine.